halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/services/v1/GenerateService.java

```diff
@@ -16,90 +16,49 @@
 
 package com.netflix.spinnaker.halyard.config.services.v1;
 
-import com.fasterxml.jackson.databind.ObjectMapper;
 import com.netflix.spinnaker.halyard.config.config.v1.AtomicFileWriter;
 import com.netflix.spinnaker.halyard.config.errors.v1.HalconfigException;
-import com.netflix.spinnaker.halyard.config.model.v1.node.Node;
-import com.netflix.spinnaker.halyard.config.model.v1.node.NodeFilter;
 import com.netflix.spinnaker.halyard.config.model.v1.node.NodeReference;
 import com.netflix.spinnaker.halyard.config.model.v1.problem.Problem;
 import com.netflix.spinnaker.halyard.config.model.v1.problem.ProblemBuilder;
-import com.netflix.spinnaker.halyard.config.spinnaker.v1.ComponentName;
-import com.netflix.spinnaker.halyard.config.spinnaker.v1.ComponentProfile;
-import lombok.extern.slf4j.Slf4j;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.stereotype.Component;
-import org.yaml.snakeyaml.Yaml;
-
+import com.netflix.spinnaker.halyard.config.spinnaker.v1.component.SpinnakerComponent;
 import java.io.IOException;
 import java.nio.file.FileSystem;
 import java.nio.file.FileSystems;
 import java.nio.file.Path;
 import java.util.ArrayList;
-import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
 
 @Component
 @Slf4j
 public class GenerateService {
   @Autowired
-  ComponentProfile componentProfile;
-
-  @Autowired
-  LookupService lookupService;
-
-  @Autowired
   String spinnakerOutputPath;
 
-  @Autowired
-  Yaml yamlParser;
-
-  @Autowired
-  ObjectMapper objectMapper;
+  @Autowired(required = false)
+  List<SpinnakerComponent> spinnakerComponents = new ArrayList<>();
 
   public void generateConfig(NodeReference nodeReference) {
-    NodeFilter filter = NodeFilter.makeAcceptAllFilter()
-        .refineWithReference(nodeReference);
-
-    List<Node> matching = lookupService.getMatchingNodesOfType(filter, Node.class);
-
-    Map<ComponentName, List<Node>> configInjectMap = new HashMap<>();
```

Comment:
This is where the halconfig was traversed to see which node to append to which component.